### PR TITLE
Implement toggleable requirement selector UI

### PR DESCRIPTION
**Pull request by SamChou19815**
### Summary 

Implement toggleable requirement UI according to [this design](https://www.figma.com/file/zHChNQWjT9RJ7K20BS28Rk/FINAL-FINAL-FINAL-SPEC?node-id=1468%3A4533). This PR only handles the UI part and stores the choice in a local state. Storing data in DB will be handled in another diff.

### Test Plan 

Test with biological science major in A&S, which is the only current requirement with toggleable requirement.
Screen Shot 2020-11-07 at 19 56 23

### CODE REVIEW

File: src/components/SubRequirement.vue [hunk 1]
```diff
@@ -82,6 +124,15 @@ export default Vue.extend({
       const type = isCompleted ? 'completed' : 'ongoing';
       this.$emit('toggleDescription', reqIndex, type, subReqIndex);
     },
+    closeMenuIfOpen() {
+      if (this.showFulfillmentOptionsDropdown) {
+        this.showFulfillmentOptionsDropdown = false;
+      }
+    },
```
Comment (benjamin-shen) [inline src/components/SubRequirement.vue:131]: Isn't this just equivalent to
```
closeMenu() {
  this.showFulfillmentOptionsDropdown = false;
}
```
That may be cleaner?

**Review by benjamin-shen — COMMENTED** [review]
Looks great, left a minor comment on the code.  
Also on the UI:
Can you make the top borders square? Also can the hover highlight not overhang the bottom of the dropdown (i.e. add border radius to the last child)? The CSS for the Add Semester dropdowns is pretty similar, if you want to reference it.

File: src/components/SubRequirement.vue (hunk 1 above)
Comment (SamChou19815) [reply src/components/SubRequirement.vue:131]: oops, right

**Review by benjamin-shen — COMMENTED** [review]
The bottom hover overhang looks fixed now. Can you remove the top border-radius too?
image
Also add cursor: pointer to everything in the dropdown

**Review by benjamin-shen — APPROVED** [review]

**Pull request merged**